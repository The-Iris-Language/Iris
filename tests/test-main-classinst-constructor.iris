.. Tests calling constructor of an object in main
class Animal () {
    public:
        int legs;
        int eyes;
        bool fur;
        
        string noise(int volume) {
            return "ruh roh raggy";
        }

    permit: 
        string color;
        bool alive;
}

class Main () {
    public:
       int univ main () {
            Animal a = new Animal();
            Olympus.println("Animal malloc'd in main");
            return 0;
       }
}