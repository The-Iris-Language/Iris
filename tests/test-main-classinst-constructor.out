Animal malloc'd in main
